src/compass/utils/geometry_utils.py

```diff
@@ -265,3 +265,162 @@ def get_unit_vector4component_of_interest(los_inc_angle, los_az_angle, comp='enu
+
+    Returns
+    -------
+    rg_arr: np.ndarray
+        Displacement in slant range direction in meters.
+    az_arr: np.ndarray
+        Displacement in azimuth direction in seconds.
+
+    Notes
+    -----
+    When `geo2rdr_params` is not provided, then the iteration
+    threshold and max # iterations are set to
+    `1.0e-8` and `25` respectively.
+
+    '''
+    if geo2rdr_params is None:
+        # default threshold and # iteration for geo2rdr
+        threshold = 1.0e-8
+        maxiter = 25
+    else:
+        threshold = geo2rdr_params.threshold
+        maxiter = geo2rdr_params.numiter
+
+    shape_arr = lon_arr.shape
+    rg_arr = np.zeros(shape_arr)
+    az_arr = np.zeros(shape_arr)
+
+    # Calculate the ENU vector in ECEF
+    for i, lon_deg in enumerate(np.nditer(lon_arr)):
+        index_arr = np.unravel_index(i, lon_arr.shape)
+        lat_deg = lat_arr[index_arr]
+        hgt = hgt_arr[index_arr]
+
+        vec_e, vec_n, vec_u = get_enu_vector_ecef(lon_deg, lat_deg)
+
+        llh_ref = np.array([np.deg2rad(lon_deg),
+                            np.deg2rad(lat_deg),
+                            hgt])
+
+        xyz_before = ellipsoid.lon_lat_to_xyz(llh_ref)
+        xyz_after_set = (xyz_before + vec_e * e_arr[index_arr]
+                              + vec_n * n_arr[index_arr]
+                              + vec_u * u_arr[index_arr])
+        llh_displaced = ellipsoid.xyz_to_lon_lat(xyz_after_set)
+
+        aztime_ref, slant_range_ref =\
+            isce3.geometry.geo2rdr(llh_ref,
+                                   ellipsoid,
+                                   orbit,
+                                   isce3.core.LUT2d(),
+                                   radargrid_ref.wavelength,
+                                   radargrid_ref.lookside,
+                                   threshold=threshold,
+                                   maxiter=maxiter)
+
+        aztime_displaced, slant_range_displaced =\
+            isce3.geometry.geo2rdr(llh_displaced,
+                                   ellipsoid,
+                                   orbit,
+                                   isce3.core.LUT2d(),
+                                   radargrid_ref.wavelength,
+                                   radargrid_ref.lookside,
+                                   threshold=threshold,
+                                   maxiter=maxiter)
+
+        rg_arr[index_arr] = slant_range_displaced - slant_range_ref
+        az_arr[index_arr] = aztime_displaced - aztime_ref
+
+    return rg_arr, az_arr
+
+
+def get_enu_vector_ecef(lon, lat, units='degree'):
+    '''
+    Calculate the east, north, and up vectors in ECEF for lon / lat provided
+
+    Parameters
+    ----------
+    lon: np.ndarray
+        Longitude of the points to calculate ENU vectors
+    lat: np.ndarray
+        Latitude of the points to calculate ENU vectors
+    units: str
+        Units of the `lon` and `lat`.
+        Acceptable unit is `radian` or `degree` (Default)
```

Comment:
```suggestion
        Acceptable unit is `radian` or `degree` (Default: degree)
```
Not sure, but I think units of angles need to be specified using the plural version?